spatialmath/baseposematrix.py

```diff
@@ -1657,7 +1657,7 @@ def _op2(left, right: Self, op: Callable):  # pylint: disable=no-self-argument
         =========   ==========   ====  ================================
 
         """
-        if isinstance(right, left.__class__):
+        if right.__class__ is left.__class__:
```

Comment:
I'm pretty sure `isinstance` is better?

Reply:
i made this change because, given the inheritance structure,
- SO3 is not an instance of SE3, but
- SE3 is an instance of SO3.
This PR seems to disallow bothse3.angdist(so3) and so3.angdist(se3).
Please feel free to make any change as necessary @bokorn-bdaii

Reply:
We could allow angdist between those too, but I am not sure if all binary operations will work here, translation based operations being computed between SO3 and SE3. That being said, it's currently only used by angdist. Depends if you want to allow cross type binary operations and put the validity checks up to future implementers.

Reply:
This is an option if we want to be able to look at angular distances between SO3 and SE3, or visa versa
```suggestion
        if isinstance(right, left.__class__) or isinstance(left, right.__class__):
```

Reply:
suggestion applied in b848e7bff7448d16af5b0c7fd3d5498c515cb7cf